ddtrace/contrib/consul/patch.py

```diff
@@ -43,9 +43,14 @@ def trace_func(wrapped, instance, args, kwargs):
             return wrapped(*args, **kwargs)
 
         path = kwargs.get('key') or args[0]
-
-        with pin.tracer.trace(name, service=pin.service, resource=path) as span:
-            span.set_tag('consul.key', path)
+        resource = name.upper()
+
+        with pin.tracer.trace(consulx.CMD, service=pin.service, resource=resource) as span:
+            rate = config.consul.get_analytics_sample_rate()
```

Comment:
nice, but can we add a test for this?

We want to test that by default this metric doesn't get set:
https://github.com/DataDog/dd-trace-py/blob/master/tests/contrib/redis/test.py#L79

But when we configure this setting, that it does:

https://github.com/DataDog/dd-trace-py/blob/master/tests/contrib/redis/test.py#L81-L103

Reply:
👍 Added tests for enabled, disabled, and w/ rate